### PR TITLE
Dev environments: rename the default rails env to "stubbed" and make the current "local" the default env

### LINKED ISSUE #5292: Dev environments: rename the default rails env to "stubbed" and make the current "local" the default env
For Queue, Hearings Prep, Case Search, Certification, and Reader we no longer need to use the stubbed environment for development. We're moving towards using FACOLS for everything, and as such we should make it the default env.

The dev will have to e.g. rename `Rails.env.local?` to `Rails.env.development?` everywhere, and rename `Rails.env.development?` to `Rails.env.stubbed?`. Also we'll have to reshuffle entries in `database.yml`. There may be other changes too.

# AC
- `rails s` should run the environment you would currently run with `rails s -e local`
- `rails s -e stubbed` should run the current default environment

### PR DESCRIPTION
**Pull request by dannysellers**
Resolves #5292

### Description
We're moving towards using FACOLS for everything, and as such we should make it the default env

### Acceptance Criteria 
- [ ] Confirm running `rails s -e stubbed` displays fake data
- [ ] Confirm running `rails s` displays FACOLS data

### Testing Plan
1. Automated tests pass in `stubbed` and `development` environments

### CODE REVIEW

**Comment by dannysellers** [comment]
Discussion thread https://dsva.slack.com/archives/C2ZAMLK88/p1525115756000641

File: Gemfile [hunk 1]
```diff
@@ -140,7 +140,7 @@ group :development, :test, :local do
   gem "rspec_junit_formatter"
 end
 
-group :development do
+group :stubbed do
```
Comment (mdbenjam) [inline Gemfile]: This should probably still apply to development as well.

File: config/initializers/react_on_rails.rb [hunk 2]
```diff
@@ -38,13 +38,13 @@
   config.prerender = false
 
   # default is true for development, off otherwise
-  config.trace = Rails.env.development?
+  config.trace = Rails.env.stubbed?
```
Comment (mdbenjam) [inline config/initializers/react_on_rails.rb]: This and the one below should probably be true if it's development as well

File: config/environments/development.rb [hunk 3]
```diff
@@ -11,24 +11,11 @@
 
   # Show full error reports and disable caching.
   config.consider_all_requests_local       = true
-
-  # Enable/disable caching. By default caching is disabled.
```
Comment (mdbenjam) [inline config/environments/development.rb]: I'm wondering if we should leave some of these configurations, not sure what all of them do, but we probably want everything that exists for development currently.

**Comment by dannysellers** [comment]
Diff 
screen shot 2018-05-07 at 2 39 29 pm

**Pull request merged**